### PR TITLE
[Web UI] Checks Page Publishing/Unpublishing

### LINKED ISSUE #1935: [Web UI] Checks page improvements
Targeted page improvements for issues discovered during QA and VQA process.

- [ ] Filter checks by subscription. (Would mirror entities page, as a user I would be somewhat conditioned to expect this.)
- [ ] If a check has been unpublished this state should be reflected in the list.
  - in particular "executed every X" should be omitted from copy to reduce any confusion.
- [ ] Add option to publish / unpublish checks.
  - Additionally it would be nice to collapse the delete action by default. We should be guiding users toward soft deletion when possible.

### PR DESCRIPTION
**Pull request by dabria**
## What is this change?
Closes #1935 
Adds the ability to publish and unpublish checks, either individually through the overflow icon on the right side, or in bulk with the checklist. A confirmation toast will appear to let the user know if the action succeeded.

screen shot 2018-09-26 at 3 33 51 pm
screen shot 2018-09-26 at 3 34 03 pm
screen shot 2018-09-26 at 3 38 48 pm

## Why is this change necessary?
Users should have the ability to [un]publish in the UI

## Does your change need a Changelog entry?
Yes, I added one.

## Do you need clarification on anything?
- We should create a new ticket for adding a graphql fragment to the CheckListHeader component.
- We should also probably allow this behaviour on the check details page
- I have not collapsed the delete button from the bulk actions header due to #2056 
- Toasts occur on success for this and execution, do we want silencing to also have a success toast?

## Were there any complications while making this change?
I had to remove defaults that were set on check creation (interval and publish). Interval was causing a bug when trying to unpublish/publish checks with cron specified.

There was discussion about adding the ability to "undo" on the notification toast. It seems rather simple to save a list of the checks interacted with, then invert them if the user clicks "undo". However, some discussion was had and it seems like the act of undoing as a whole might want to be implemented in a smarter way. Perhaps deleting should also allow for undoing. There's a conversation to be had there, and for now I chose to not implement the behaviour.

### CODE REVIEW

File: dashboard/src/components/partials/ChecksList/ChecksListItem.js [hunk 1]
```diff
@@ -87,25 +90,12 @@ class CheckListItem extends React.Component {
...
-                  )}
-                </strong>{" "}
-                by{" "}
-                <strong>
-                  {check.subscriptions.length}{" "}
-                  {check.subscriptions.length === 1
-                    ? "subscription"
-                    : "subscriptions"}
-                </strong>.
+                {check.publish && <CheckPublishInfo check={check} />}
```
Comment (jamesdphillips) [inline dashboard/src/components/partials/ChecksList/ChecksListItem.js]: 🤔 how would you feel about naming this component CheckSummary or CheckDescriptor and including the unpublished case as well?

File: dashboard/src/components/partials/ChecksList/CheckPublishInfo.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  };
+
+  render() {
+    const { check } = this.props;
+
+    return (
+      <React.Fragment>
+        Executed{" "}
+        <strong>
+          {check.interval ? (
```
Comment (jamesdphillips) [inline dashboard/src/components/partials/ChecksList/CheckPublishInfo.js]: Ideally this component would include a query fragment describing the fields it requires. In this case it's referencing the `interval`, `cron`, and `subscription` fields.

File: dashboard/src/components/partials/ChecksList/ChecksListHeader.js [hunk 3]
```diff
@@ -9,24 +9,30 @@ import ListSortMenu from "/components/partials/ListSortMenu";
...
     onClickDelete: PropTypes.func.isRequired,
     onClickExecute: PropTypes.func.isRequired,
+    onClickSetPublish: PropTypes.func.isRequired,
     onClickSilence: PropTypes.func.isRequired,
     rowCount: PropTypes.number.isRequired,
     selectedItems: PropTypes.array.isRequired,
     toggleSelectedItems: PropTypes.func.isRequired,
   };
 
+  // TODO add a fragment
```
Comment (jamesdphillips) [inline dashboard/src/components/partials/ChecksList/ChecksListHeader.js]: 😬 my bad

File: dashboard/src/components/relocation/PublishCheckStatusToast.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  }
+
+  render() {
+    const { mutation, onClose, checkName, entityName } = this.props;
+    const { loading } = this.state;
+
+    const subject = (
+      <React.Fragment>
+        <strong>{checkName}</strong>
+        {entityName && (
```
[the remark this answers is not in the document]
Comment (dabria) [reply dashboard/src/components/relocation/PublishCheckStatusToast.js]: Good catch. Remnants of copying the Execute toast

File: dashboard/src/components/partials/ChecksList/ChecksListItem.js (hunk 1 above)
Comment (dabria) [reply dashboard/src/components/partials/ChecksList/ChecksListItem.js]: Seems like a good idea, I'll add that.

File: backend/apid/graphql/schema/mutations.graphql [hunk 5]
```diff
@@ -87,7 +87,7 @@ input CheckConfigInputs {
   command: String
 
   "interval is the time interval, in seconds, in which the check should be run. Defaults to 60."
-  interval: Int = 60
+  interval: Int
```
[the remark this answers is not in the document]
Comment (dabria) [reply backend/apid/graphql/schema/mutations.graphql]: Good catch!

**Pull request merged**